### PR TITLE
Initial unit tests to test generate schedule resolvers

### PR DESCRIPTION
**Pull request by AntonNiko**
This PR adds 3 unit tests, didn't want to overwhelm with large PR so want to make sure this format works for subsequent unit tests.

### CODE REVIEW

File: src/algorithm/index.ts [hunk 1]
```diff
@@ -49,17 +49,22 @@ const algo2c4 = createPredict(
   })
 );
 
+// Set of endpoints returned when calling request for company 3 or company 4 algorithms.
```
Comment (AntonNiko) [inline src/algorithm/index.ts:52]: Extracting these objects allows the unit tests to spy on them and validate the input we send to the endpoint.

**Review by aqeelmozumder — COMMENTED** [review]
I believe the snapshots files do not require to be in the PR, as they will be generated automatically.

**Comment by AntonNiko** [comment]
@aqeelmozumder Should we add `.snap` to the .gitignore? Can add it on

**Comment by aqeelmozumder** [comment]
> @aqeelmozumder Should we add `.snap` to the .gitignore? Can add it on

Yes, that would be great

**Comment by aomi** [comment]
@aqeelmozumder @AntonNiko the snapshot tests you generate while developing and commit them to version control.
That being said, snapshot tests might not be the best suited for our tests because the values we get from each respective service might not be exactly equal as we expect.

**Comment by aqeelmozumder** [comment]
> @aqeelmozumder @AntonNiko the snapshot tests you generate while developing and commit them to version control. That being said, snapshot tests might not be the best suited for our tests because the values we get from each respective service might not be exactly equal as we expect.

The Snapshots will always be updated if there are changes and the tests will compare with the up to date snaps. Even if we commit the snap files, the snap files will most likely to get replaced with an updated snaps when the CI runs.

**Pull request closed without merging**